src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -367,9 +367,16 @@ public void periodic() {
   }
 
   private void updateOdometry() {
+    for (int i = 0; i < modules.length; i++) {
+      Logger.recordOutput(
+          "Swerve/Updates Since Last " + i, modules[i].getOdometryTimestamps().length);
+    }
     double[] sampleTimestamps =
         modules[0].getOdometryTimestamps(); // All signals are sampled together
-    int sampleCount = sampleTimestamps.length;
+    int sampleCount =
+        Math.min(
+            Arrays.stream(modules).mapToInt(m -> m.getOdometryTimestamps().length).min().getAsInt(),
```

Comment:
What guarantee is there that two modules with the same number of results have the same sampled times? Do they need to?

Reply:
they should as long as neither is taking the earliest element in the queue, since they should both get the same earliest sample and only use the samples both have.

Reply:
What code makes them only use samples in common?

Reply:
the more i think it through the more edge cases that could break it i think of, ill just do it properly